libglusterfs/src/common-utils.c

```diff
@@ -649,6 +649,9 @@ gf_print_trace(int32_t signum, glusterfs_ctx_t *ctx)
 
     gf_log_disable_suppression_before_exit(ctx);
 
+    if (!ctx || ctx->log.logger != gf_logger_glusterlog)
```

Comment:
I don't think it will work always. As per the issue (https://github.com/gluster/glusterfs/issues/3882) it seems you have mounted a volume without enabling syslog logger and the process was hanged, by default gluster_log is enabled and in case of default also we do send critical messages to syslog.I think we have to ignore sending a message syslog completely to avoid an issue.

Reply:
@mohit84 ,I noticed that in default case syslog will also be called at GF_LOG_ALERT level even though the logger is set as gf_logger_glusterlog. 

```
case gf_logger_glusterlog:
            pthread_mutex_lock(&log->logfile_mutex);
            {
                if (log->logfile) {
                    fprintf(log->logfile, "%s\n", msg);
                    fflush(log->logfile);
                } else {
                    fprintf(stderr, "%s\n", msg);
                    fflush(stderr);
                }

#ifdef GF_LINUX_HOST_OS
                /* We want only serious logs in 'syslog', not our debug
                 * and trace logs */
                if (level && log->gf_log_syslog &&
                    (level <= log->sys_log_level))
                    syslog((level - 1), "%s\n", msg);
#endif
            }
            pthread_mutex_unlock(&log->logfile_mutex);

            break;
```

So, can we temporarily disable log->gf_log_syslog in gf_print_trace to keep syslog from calling?

Reply:
I don't think we should avoid the log, the messages are important. We should avoid only syslog call. 

@xhernandez 
  What do you think?

Reply:
> I don't think we should avoid the log, the messages are important. We should avoid only syslog call.

By considering this, can we temporarily disable log->gf_log_syslog in gf_print_trace to keep syslog from calling? These important messages can be safely printed when logger is set as gluster_log.

Reply:
Yes i was expecting the same.

Reply:
Actually there are a lot of non-async-signal-safe functions used in signal handlers. I'm ok to temporarily disable syslog here, but it's not a complete solution. I'm not really sure how to fix this in a good and complete way.

For now, we can fix this particular issue.